common/djangoapps/enrollment/views.py

```diff
@@ -700,3 +707,102 @@ def post(self, request):
                 )
+
+
+@can_disable_rate_limit
+class CourseEnrollmentsApiListView(DeveloperErrorViewMixin, ListAPIView):
+    """
+        **Use Cases**
+
+            Get a list of all course enrollments given a course ID or list of usernames.
+
+        **Example Requests**
+
+            GET /api/enrollment/v1/enrollments?course_id={course_id}
+
+            GET /api/enrollment/v1/enrollments?username={username},{username},{username}
+
+            GET /api/enrollment/v1/enrollments?course_id={course_id}&username={username}
+
+        **Query Parameters for GET**
+
+            * course_id: Filters the result to course enrollments for the course corresponding to the
+              given course ID. The value must be URL encoded. Optional. Required if the 'username' parameter
+              is omitted.
+
+            * username: List of comma-separated usernames. Filters the result to the course enrollments
+              of the given users. Optional. Required if the 'course_id' parameter is omitted.
+
+            * page_size: Number of results to return per page.
+
+            * page: Page number to retrieve.
+
+        **Response Values**
+
+            If the request for information about the course enrollments is successful, an HTTP 200 "OK" response
+            is returned.
+
+            The HTTP 200 response has the following values.
+
+            * count: The number of course enrollments matching the request.
+
+            * num_pages: The total number of pages in the result.
+
+            * current_page: The current page number.
+
+            * results: A list of the course enrollments matching the request.
+
+                * created: Date and time when the course enrollment was created.
+
+                * mode: Mode for the course enrollment.
+
+                * is_active: Whether the course enrollment is active or not.
+
+                * user: Username of the user in the course enrollment.
+
+                * course_id: Course ID of the course in the course enrollment.
+
+            * next: The URL to the next page of results, or null if this is the
+              last page.
+
+            * start: The list index of the first item in the response.
+
+            * previous: The URL to the next page of results, or null if this
+              is the first page.
+
+            If the user is not logged in, a 401 error is returned.
+
+            If the user is not global staff, a 403 error is returned.
+
+            If the specified course_id is not valid or any of the specified usernames
+            are not valid, a 400 error is returned.
+
+            If the specified course_id does not correspond to a valid course or if all the specified
+            usernames do not correspond to valid users, an HTTP 200 "OK" response is returned with an
+            empty 'results' field.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication,
+                              SessionAuthentication,)
```

Comment:
These should be ``OAuth2AuthenticationAllowInactiveUser`` and ``SessionAuthenticationAllowInactiveUser``

Reply:
@xitij2000, is there any reason to allow inactive users to access this API?

Reply:
@lgp171188 You can look up where those classes are defined and read up why in details, but the gist is:
* We will be accessing this as a special robot API user that no one is supposed to log in as.
* Open edX uses two auth libraries and the class override has special logic to support both.

Reply:
Thanks for catching that @xitij2000  :)